File: users-api/src/main/java/net/plshark/errors/BadRequestException.java
package net.plshark.errors;

/**
 * Exception indicating a request was invalid
 */
public class BadRequestException extends RuntimeException {

    private static final long serialVersionUID = 5359694168645391199L;

    /**
     * Create a new instance
     * @param message the detail message
     */
    public BadRequestException(String message) {
        super(message);
    }

    /**
     * Create a new instance
     * @param message the detail message
     * @param cause the cause
     */
    public BadRequestException(String message, Throwable cause) {
        super(message, cause);
    }
}

File: users-api/src/main/kotlin/net/plshark/errors/BadRequestException.kt
package net.plshark.errors

/**
 * Exception indicating a request was invalid
 */
class BadRequestException : RuntimeException {
    /**
     * Create a new instance
     * @param message the detail message
     */
    constructor(message: String?) : super(message)

    /**
     * Create a new instance
     * @param message the detail message
     * @param cause the cause
     */
    constructor(message: String?, cause: Throwable?) : super(message, cause)
}